src/Parser/Hunter/Marksmanship/Modules/Talents/AMurderOfCrows.js

```diff
@@ -7,27 +7,51 @@ import SPELLS from 'common/SPELLS';
 import StatisticBox from "Main/StatisticBox";
 import SpellIcon from "common/SpellIcon";
 import { formatNumber } from "common/format";
+import SpellLink from "common/SpellLink";
+
+//generally accepted rule is to save crows if boss is below 25% health.
+const CROWS_SAVE_PERCENT = 0.25;
+const EXECUTE_PERCENT = 0.2;
 
 class AMurderOfCrows extends Analyzer {
 
   static dependencies = {
     combatants: Combatants,
   };
 
+  bossIDs = [];
   damage = 0;
+  shouldHaveSaved = 0;
 
   on_initialized() {
     this.active = this.combatants.selected.hasTalent(SPELLS.A_MURDER_OF_CROWS_TALENT_SHARED.id);
+    this.owner.report.enemies.forEach(enemy => {
+      enemy.fights.forEach(fight => {
+        if (fight.id === this.owner.fight.id && enemy.type === "Boss") this.bossIDs.push(enemy.id);
+      });
+    });
   }
 
-  //TODO: Add some sort of HP checker for the casts to ensure it isn't cast at below 25% hp (maybe)
   on_byPlayer_damage(event) {
     const spellId = event.ability.guid;
     if (spellId !== SPELLS.A_MURDER_OF_CROWS_SPELL.id) {
       return;
     }
     this.damage += event.amount;
   }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.A_MURDER_OF_CROWS_SPELL.id) {
+      return;
+    }
+    if (event.targetInstance === undefined && event.maxHitPoints && this.bossIDs.indexOf(event.targetID) !== -1) {
```

Comment:
Why checking for no targetInstance and also if target is a boss? Are there any bosses with more than one instance?

Also, the last condition should be `this.bossIDs.includes(event.targetID)`

Reply:
Checks if target is boss to ensure it doesn't give false flags if the player is using on an add that has between that amount of health (which could be fine - especially if the mob then dies within 15 seconds so that crows is reset) 

Should probably work without the targetInstance check 🤔 I shamelessly stole it from ``Parser/Hunter/Marksmanship/Modules/Traits/Bullseye.js`` where part of this code was originally in until we figured out how we wanted to do it